### PR TITLE
Add a double-valued summary type of MetricData.

### PR DESCRIPTION
**Pull request by jkwatson**
Step 1 in creating a MinMaxSumCount aggregation

### CODE REVIEW

File: sdk/src/main/java/io/opentelemetry/sdk/metrics/data/MetricData.java [hunk 1]
```diff
@@ -152,6 +153,74 @@ public static DoublePoint create(
...
+        double sum,
+        List<DoubleValueAtPercentile> percentileValues) {
+      return new AutoValue_MetricData_SummaryPoint(
+          startEpochNanos, epochNanos, labels, count, sum, percentileValues);
+    }
+  }
+
+  @Immutable
+  @AutoValue
+  public abstract static class DoubleValueAtPercentile {
```
Comment (bogdandrutu) [inline sdk/src/main/java/io/opentelemetry/sdk/metrics/data/MetricData.java]: s/DoubleValueAtPercentile/ValueAtPercentile?

File: sdk/src/main/java/io/opentelemetry/sdk/metrics/data/MetricData.java [hunk 2]
```diff
@@ -196,6 +266,13 @@ public static DoublePoint create(
        * @since 0.1.0
        */
       MONOTONIC_DOUBLE,
+
+      /**
+       * A Summary of measurements of numeric values, containing the minimum value recorded, the
+       * maximum value recorded, the sum of all measurements and the total number of measurements
+       * recorded.
+       */
+      DOUBLE_SUMMARY,
```
Comment (bogdandrutu) [inline sdk/src/main/java/io/opentelemetry/sdk/metrics/data/MetricData.java]: s/DOUBLE_SUMMARY/SUMMARY?

File: sdk/src/main/java/io/opentelemetry/sdk/metrics/data/MetricData.java [hunk 3]
```diff
@@ -152,6 +153,74 @@ public static DoublePoint create(
...
+
+    /**
+     * Percentile values in the summarization. Note: a percentile 0.0 represents the minimum value
+     * in the distribution.
+     *
+     * @return the percentiles values.
+     */
+    public abstract List<DoubleValueAtPercentile> getPercentileValues();
+
+    public static SummaryPoint create(
```
Comment (bogdandrutu) [inline sdk/src/main/java/io/opentelemetry/sdk/metrics/data/MetricData.java]: A small test similar to metricData_LongPoints would be good.

File: sdk/src/main/java/io/opentelemetry/sdk/metrics/data/MetricData.java (hunk 1 above)
Comment (jkwatson) [reply sdk/src/main/java/io/opentelemetry/sdk/metrics/data/MetricData.java]: I'm cool either way.  👍

**Comment by bogdandrutu** [comment]
Nit: @jkwatson please try to create PR in branches that are in your own clone of the repo not directly in this repo :)

**Comment by jkwatson** [comment]
> Nit: @jkwatson please try to create PR in branches that are in your own clone of the repo not directly in this repo :)

oh shoot. that was 100% not intentional. I'm actually not even sure how I did that. I'll close and re-open.

**Comment by jkwatson** [comment]
re-opening against fork branch

**Pull request closed without merging**